frontend/src/metabase/dashboard/actions/data-fetching.js

```diff
@@ -345,7 +345,16 @@ export const fetchCardData = createThunkAction(
 
       cancelFetchCardData(card.id, dashcard.id);
 
-      if (clearCache) {
+      // When dashcard parameters change, we need to clean previous (stale)
+      // state so that the loader spinner shows as expected (#33767)
+      const hasParametersChanged =
+        !lastResult ||
```

Comment:
@alxnddr do you think it should be this or `lastResult && equals(...)`?

Reply:
I think the equals part should definitely be `!equals(...)` since there should be difference. Regarding lastResult, I think we do not need to clear anything if there is no last result. But these are guesses based on logic which may not always work 😬